course-v2/layouts/partials/desktop_course_info.html

```diff
@@ -1,6 +1,8 @@
 <div class="border-left">
-  <div id="desktop-course-info">
-    {{ partial "course_info.html" (dict "context" . "inPanel" true) }}
+  {{/*  adding d-none class here as this section (with id="desktop-course-drawer") is being controlled (shown/hidden) 
+  via desktop_course_drawer.ts hence hidden at the start to avoid a glitch  */}}
```

Comment:
I don't think that this is accomplishing what you're trying to do here.  I get some layout thrashing upon refresh.  setting this div to `d-none` does indeed hide the div, but the parent element calling this partial places it inside a bootstrap column with `col-3` set on it with gives that section width regardless of its contents.  We should try and figure out a way to set these classes before the DOM loads to avoid layout thrashing.  One approach could be to inline the script as far up in the page as possible at the start of the `body` tag, rather than including it in the bundle.  If we do this and make the script pure JS (not using any libraries) then it should execute before anything else.

Here's a screen recording that demonstrates this: https://nextcloud.c4103.com/index.php/s/PkBpXfx4j6SmK8H

Reply:
@gumaerc Thanks for taking a look into this.
`d-none` is applied to hide the element when the HTML is being rendered so when JS is loaded (after a couple of milliseconds maybe) it shows/hides the drawer according to the user's preference stored in `local storage`. If we don't apply `d-none` in the first place and the user has hidden drawer then the drawer is first visible and then hidden when the JS is run which looked pretty bad. 
But I agree that the `col-3` element should also be not shown when the drawer is not showing so I guess I'll make this update to hide the complete section including the outer/parent, `col-3` element. Does this sound fine to you? 

For layout thrashing or I'd say the glitch: I also had this in notice but was not sure how much noticeable it is for the end users so I sent a video to @pdpinch in order to take his thoughts and we came to this conclusion that it's not too much noticeable but I think Peter will be in a better position to tell once he experiences it himself.

I have already tried the approach of inline vanilla JS but it also had some issues and was not working perfectly. If you feel that this issue needs to be catered then I think I'll give this inline approach another shot and see how things go. If things don't go well, then we'll have to come up with a different approach.

Reply:
@gumaerc I tried inline JS (moved all the JS code) in `baseof.html` and it works fine. Now I'll update the PR shortly and it will be up for review again. 
Thanks